api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -35,4 +38,9 @@ message QuicProtocolOptions {
   // Runtime flag that controls whether the listener is enabled or not. If not specified, defaults
   // to enabled.
   core.v3.RuntimeFeatureFlag enabled = 4;
+
+  // A multiplier to number of connections which is used to determine how many packets to read per event loop. A reasonable ratio should allow the listener to process enough payload but not starve TCP and other UDP sockets and also prevent long event loop duration.
+  // The default value is 32. This means if there are N QUIC connections, the total number of packets to read in each READ event will be 32 * N.
+  google.protobuf.UInt32Value packets_received_to_connection_count_ratio = 5
+      [(validate.rules).uint32 = {gte: 1}];
```

Comment:
Do we also want to make the global max packets per wakeup limit configurable?

If yes, future PR.

Reply:
I'm not sure about that. I doubt that we should encourage users to tune the quic ratio or even the global max packets per wakeup, because it's not just starving other sockets, but also delaying callbacks and timers for the next event loop. So maybe we can make it const for now, and see if anyone would request to make it configurable?